### PR TITLE
Remove deprecated options & set default value for ec2 ami list owner option

### PR DESCRIPTION
**Pull request by vsingh-msys**
### Description

 - Removed options `--tag-node-in-chef` and `--security-group-ids`
 - As `--tags` also deprecated but available in chef-core so `--tags` removed from knife-ec2.
 - Currently `--tags` and `--chef-tag` do the same things so we can also deprecated `--chef-tag` from knife-ec2 plugin?

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [x] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by btm** [comment]
we can deprecate `--chef-tag` but in the message we need to note that currently `--tags` in core takes this format: `--tags foo,bar,zap` while we had moved `--chef-tag` to use this format: `--chef-tag foo --chef-tag bar --chef-tag zap`...


at some point we should switch code to use the other pattern, but for now the deprecation message should point out the difference.

**Comment by vsingh-msys** [comment]
Hey @btm Thanks for reviews comment I have deprecated `--chef-tag` and added deprecation warning `[DEPRECATED] --chef-tag option is deprecated and will be removed in future release. Use --tags TAGS option instead.`
could you please have a look again? let me know if anything is missing.
Thnaks

File: lib/chef/knife/ec2_server_create.rb [hunk 1]
```diff
@@ -753,17 +719,13 @@ def plugin_validate_options!
...
         end
 
-        if config_value(:tag_node_in_chef)
-          ui.warn("[DEPRECATED] --tag-node-in-chef option is deprecated. Use --chef-tag option instead.")
-        end
-
-        if config_value(:tags)
-          ui.warn("[DEPRECATED] --tags option is deprecated. Use --aws-tag option instead.")
+        if config_value(:chef_tag)
+          ui.warn("[DEPRECATED] --chef-tag option is deprecated and will be removed in future release. Use --tags TAGS option instead.")
```
Comment (Nimesh-Msys) [inline lib/chef/knife/ec2_server_create.rb]: I think `deprecated_options` could be used to handle this

Comment (vsingh-msys) [reply lib/chef/knife/ec2_server_create.rb]: @Nimesh-Msys As `--chef-tag` is used multiple times for multiple values and proc helper is used to storing these values in the array. 
But  `deprecated_option` don't have any proc option as of now.

We might think of improving mixlib-cli to support `value_mapper` with  `proc` block as well.

**Pull request merged**